examples/opentelemetry-otlp.rs

```diff
@@ -108,20 +108,24 @@ fn init_tracer() -> Tracer {
         .with_batch_config(BatchConfig::default())
         .with_exporter(opentelemetry_otlp::new_exporter().tonic())
         .install_batch(runtime::Tokio)
-        .unwrap()
+        .unwrap();
+
+    global::set_tracer_provider(provider.clone());
```

Comment:
Yes, as I understand this is equivalent to what was there before. See the opentelemetry-otlp [changelog](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/CHANGELOG.md#v0170):

> Breaking The methods OtlpTracePipeline::install_simple() and OtlpTracePipeline::install_batch() would now return TracerProvider instead of Tracer. These methods would also no longer set the global tracer provider. It would now be the responsibility of users to set it by calling global::set_tracer_provider(tracer_provider.clone());. Refer to the [basic-otlp](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/examples/basic-otlp/src/main.rs) and [basic-otlp-http](https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-otlp/examples/basic-otlp-http/src/main.rs) examples on how to initialize OTLP Trace Exporter.

(We could of course restructure how that code is put together to make it more obvious?)

Reply:
> Is this change somehow necessary due to the upgrade? This could be a surprising side-effect.

The rationale behind these changes are well summarized in https://github.com/open-telemetry/opentelemetry-rust/issues/1592 and https://github.com/open-telemetry/opentelemetry-rust/issues/1810. But we can discuss more here if required.

Reply:
Ah, makes sense.

I don't think any restructuring is really necessary, I just saw a new side effect but it's just been moved here from within another function so it's fine by me.

On mostly unrelated note, in the changelog you linked, they also mention bumping MSRV to 1.70. `tracing-opentelemetry` has 1.65. It's not a direct dependency so it doesn't force this crate to update it, but I'd expect most users would want to use an `otlp` exporter so it might be worth at least considering having the same one? Feel free to ignore this though.

Reply:
I don't think we should bump this crate's MSRV just because downstream users probably want to take an additional dependency on another crate that has a higher MSRV. So I'll just keep this for now.